agent/log-drop/log-drop.go

```diff
@@ -0,0 +1,70 @@
+package log_drop
+
+import (
+	"context"
+)
+
+type Level int
+
+const (
+	TRACE Level = iota
+	DEBUG
+	INFO
+	WARN
+	ERROR
+)
+
+//go:generate mockery --name Logger --inpackage
+type Logger interface {
+	Info(string, ...interface{})
+}
+
+type Log struct {
+	s string
+	i []interface{}
+	l Level
+}
+
+type logDrop struct {
+	logger Logger
+	logCH  chan Log
+	name   string
+	dropFn func(l Log)
+}
+
+func (r *logDrop) Info(s string, i ...interface{}) {
+	r.pushLog(Log{l: INFO, s: s, i: i})
+}
+
+func (r *logDrop) pushLog(l Log) {
+	select {
+	case r.logCH <- l:
+	default:
+		r.dropFn(l)
+	}
+}
+
+func (r *logDrop) logConsumer(ctx context.Context) {
+	for {
+		select {
+		case l := <-r.logCH:
+			switch l.l {
+			case INFO:
+				r.logger.Info(l.s, l.i)
+			}
+		case <-ctx.Done():
+			return
+		}
+	}
+}
+
+func NewLogDrop(ctx context.Context, name string, depth int, logger Logger, dropFn func(l Log)) Logger {
```

Comment:
Could we add some godoc comments here too please? 🙇🏻‍♂️ I think it's particularly important to point out that it starts a goroutine which needs to be stopped by canceling the given context.